sys/cddl/dev/dtrace/dtrace_ioctl.c

```diff
@@ -22,12 +22,26 @@
  *
  */
 
+#include <dtrace_ioctl_compat.c>
+
 static int dtrace_verbose_ioctl;
 SYSCTL_INT(_debug_dtrace, OID_AUTO, verbose_ioctl, CTLFLAG_RW,
     &dtrace_verbose_ioctl, 0, "log DTrace ioctls");
 
 #define DTRACE_IOCTL_PRINTF(fmt, ...)	if (dtrace_verbose_ioctl) printf(fmt, ## __VA_ARGS__ )
 
+#ifdef COMPAT_FREEBSD64
+#define SIZEOF(BASE_TYPE) \
+	(SV_CURPROC_FLAG(SV_CHERI) ? sizeof (BASE_TYPE##_t) : \
+				     sizeof (BASE_TYPE##64_t) )
+#define OFFSETOF(BASE_TYPE,FIELD) \
+	(SV_CURPROC_FLAG(SV_CHERI) ? offsetof(BASE_TYPE##_t, FIELD) : \
+				     offsetof(BASE_TYPE##64_t, FIELD) )
+#else
+#define SIZEOF(BASE_TYPE) (sizeof (BASE_TYPE##_t))
+#define OFFSETOF(BASE_TYPE,FIELD) offsetof(BASE_TYPE##_t, FIELD)
+#endif
```

Comment:
I don't particularly like these macros, but:
1. They should have a DTRACE_ prefix
2. You need spaces after your commas
3. Macro arguments should be follow normal function argument case conventions

Reply:
If you don't like these macros, how would you have done?

Reply:
I think that omitting the `DTRACE` prefix makes the diff nicer. Otherwise many lines need to be broken, and why do we really need this prefix here? Maybe it's simply better to undefine them? Similar things are done in many places